src/radical/pilot/agent/staging_input/default.py

```diff
@@ -151,17 +157,26 @@ def work(self, cu):
                 if   directive['action'] == rpc.LINK: os.symlink     (source, abs_target)
                 elif directive['action'] == rpc.COPY: shutil.copyfile(source, abs_target)
                 elif directive['action'] == rpc.MOVE: shutil.move    (source, abs_target)
-                else:
-                    # FIXME: implement TRANSFER mode
-                    raise NotImplementedError('Action %s not supported' % directive['action'])
+                elif directive['action'] == rpc.TRANSFER:
+                    srm_dir = saga.filesystem.Directory('srm://proxy/?SFN=bogus')
+                    if srm_dir.exists(source):
```

Comment:
Hmm, I am not sure I understand that part.  I can see how that works for SRM - but it immediately excludes other URLs which are otherwise valid, and could be transferred via SAGA.  Wouldn't it make sense to here switch for the schema?  (and additionally, to get really fancy, and to include a resource config option for an SRM proxy?)